storage/memory/peer_store.go

```diff
@@ -92,6 +92,14 @@ func (s *peerStore) shardIndex(infoHash bittorrent.InfoHash, p bittorrent.Peer)
 	return idx
 }
 
+func (s *peerStore) shardIndexByAddressType(infoHash bittorrent.InfoHash, v6 bool) uint32 {
```

Comment:
Couldn't we merge this with the function above and and put `len(p.IP) == net.IPv6len` in all the call sites?

Also, I'm not totally positive that I understand how incrementing the index this way avoids collisions.